models/shopcart.py

```diff
@@ -33,6 +33,16 @@ def delete(self):
         """ Deletes a Shopcart in the database """
         Shopcart.__data.remove(self)
 
+    def serialize(self):
+        """ Serializes a shopcart into a dictionary """
+        return {"uid": self.uid, "products": self.products }
+
+    def get_available_id(self):
```

Comment:
We should never need this as we're using a `user_id`. Meaning that we'll never automatically assign an `id` but require a valid one.

Reply:
Done @thoughts1053 !